web/client-api/src/main/java/io/deephaven/web/client/api/tree/JsTreeTable.java

```diff
@@ -676,38 +706,41 @@ public void setExpanded(Object row, boolean isExpanded) {
 
         final TreeRow r;
         if (row instanceof Double) {
-            r = currentViewportData.rows.getAt((int) ((double) row - lastResult.getSnapshotStart()));
+            r = currentViewportData.rows.getAt((int) ((double) row - currentViewportData.offset));
         } else if (row instanceof TreeRow) {
             r = (TreeRow) row;
         } else {
             throw new IllegalArgumentException("row parameter must be an index or a row");
         }
 
-        Key myRowKey = r.myKey();
-        Key parentRowKey = r.parentKey();
-        JsLog.debug("setExpanded enqueued");
-        // With the keys collected for the currently-visible item to expand/collapse, we can enqueue an operation
-        // to modify that node
-        enqueue(isExpanded ? TreeTableRequest.TreeRequestOperation.Expand
-                : TreeTableRequest.TreeRequestOperation.Contract, () -> {
-                    TreeNodeState node = expandedMap.get(parentRowKey);
-                    if (node == null) {
-                        throw new IllegalStateException("Parent isn't available, can't manipulate child: setExpanded("
-                                + row + ", " + isExpanded + ")");
-                    }
-                    if (isExpanded) {
-                        node.expand(myRowKey);
-                    } else {
-                        node.collapse(myRowKey);
-                    }
-                });
-        scheduleSnapshotQuery(true);
+        r.appendKeyData(keyTableData, isExpanded);
+        if (keyTable != null) {
+            keyTable.then(t -> {
+                t.close();
+                return null;
+            });
+            keyTable = null;
+        }
+
+        if (viewTicket != null) {
+            viewTicket.release();
+            viewTicket = null;
+        }
```

Comment:
`viewTicket`, `filteredTable`, `sortedTable` all have blocks like this that are in applyFilter, applySort, close; could refactor them into private methods `releaseViewTicket`, `releaseFilteredTable`, `releaseSortedTable` to clean that up a bit.

Reply:
Yeah, I feel like I'm still going to hate it, since there is a "chain" of them, and depending on how far up the "chain" you are, the rest need to be released too. I'm going to see about an enum instead.